common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -76,7 +76,10 @@ public OneForOneStreamManager() {
   @Override
   public ManagedBuffer getChunk(long streamId, int chunkIndex) {
     StreamState state = streams.get(streamId);
-    if (chunkIndex != state.curChunk) {
+    if (state == null) {
+      throw new IllegalStateException(String.format(
+        "Requested chunk has been removed for streamId %s", streamId));
```

Comment:
Nit: Can we say that `chunk not available since streamId %s is closed". This seems more accurate

Reply:
done